src/main/java/core/basesyntax/strategy/ReturnHandler.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.FruitTransaction;
+import java.util.List;
+
+public class ReturnHandler implements OperationHandler {
+    private static final String RETURN = "r";
+
+    @Override
+    public List<FruitTransaction> getFruitFinalQuantity(List<FruitTransaction> listOfAll,
+                                                        List<FruitTransaction> listOfFruits) {
+        for (FruitTransaction listOfFruit : listOfFruits) {
+            int oldQuantity = 0;
+            for (FruitTransaction fruitTransaction : listOfAll) {
+                if (listOfFruit.getFruit().equals(fruitTransaction.getFruit())
+                        && fruitTransaction.getType().equals(RETURN)) {
+                    int quantity = fruitTransaction.getQuantity();
+                    listOfFruit.setQuantity(quantity + oldQuantity);
+                    oldQuantity = quantity;
```

Comment:
The nested loop here is inefficient as it results in an O(n^2) complexity. If the list of transactions is large, this could lead to performance issues. Consider using a Map to track the quantities of fruits, which would allow for O(1) access times when updating quantities.